node/narwhal/src/primary.rs

```diff
@@ -1053,3 +1053,317 @@ impl<N: Network> Primary<N> {
         self.gateway.shut_down().await;
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::MAX_EXPIRATION_TIME_IN_SECS;
+
+    use snarkos_node_narwhal_committee::MIN_STAKE;
+    use snarkos_node_narwhal_ledger_service::MockLedgerService;
+
+    use bytes::Bytes;
+    use rand::RngCore;
+
+    type CurrentNetwork = snarkvm::prelude::Testnet3;
+
+    // Returns a primary and a list of accounts in the configured committee.
+    async fn primary_without_handlers(
+        rng: &mut TestRng,
+    ) -> (Primary<CurrentNetwork>, Vec<(SocketAddr, Account<CurrentNetwork>)>) {
+        // Create a committee containing the primary's account.
+        let (accounts, committee) = {
+            const COMMITTEE_SIZE: usize = 4;
+            let mut accounts = Vec::with_capacity(COMMITTEE_SIZE);
+            let mut members = IndexMap::new();
+
+            for i in 0..COMMITTEE_SIZE {
+                let socket_addr = format!("127.0.0.1:{i}").parse().unwrap();
+                let account = Account::new(rng).unwrap();
+                members.insert(account.address(), MIN_STAKE);
+                accounts.push((socket_addr, account));
+            }
+
+            (accounts, Committee::<CurrentNetwork>::new(1, members).unwrap())
+        };
+
+        let account = accounts.first().unwrap().1.clone();
+        let storage = Storage::new(committee, 10);
+        let ledger = Arc::new(MockLedgerService::new());
+
+        // Initialize the primary.
+        let mut primary = Primary::new(account, storage, ledger, None, None).unwrap();
+
+        // Construct a worker instance.
+        primary.workers = Arc::from([Worker::new(
+            0, // id
+            Arc::new(primary.gateway.clone()),
+            primary.storage.clone(),
+            primary.ledger.clone(),
+            primary.proposed_batch.clone(),
+        )
+        .unwrap()]);
+
+        (primary, accounts)
+    }
+
+    // Creates a mock solution.
+    fn sample_unconfirmed_solution(
+        rng: &mut TestRng,
+    ) -> (PuzzleCommitment<CurrentNetwork>, Data<ProverSolution<CurrentNetwork>>) {
+        // Sample a random fake puzzle commitment.
+        let affine = rng.gen();
+        let commitment = PuzzleCommitment::<CurrentNetwork>::from_g1_affine(affine);
+        // Sample random fake solution bytes.
+        let mut vec = vec![0u8; 1024];
```

Comment:
```suggestion
        let mut vec = vec![0u8; 1024];
```
should we perhaps have different lengths here, just in case we were to hit some edge case?

Reply:
done